lib/identity_cache/cached/reference/has_many.rb

```diff
@@ -49,28 +49,33 @@ def clear(record)
         end
 
         def fetch(records)
-          fetch_embedded(records)
+          fetch_async(LoadStrategy::Eager, records) { |child_records| child_records }
+        end
 
-          ids_to_parent_record = records.each_with_object({}) do |record, hash|
-            child_ids = record.send(cached_ids_name)
-            child_ids.each do |child_id|
-              hash[child_id] = record
+        def fetch_async(load_strategy, records)
+          fetch_embedded_async(load_strategy, records) do
+            ids_to_parent_record = records.each_with_object({}) do |record, hash|
+              child_ids = record.send(cached_ids_name)
+              child_ids.each do |child_id|
+                hash[child_id] = record
+              end
             end
-          end
 
-          parent_record_to_child_records = Hash.new { |h, k| h[k] = [] }
+            load_strategy.load_multi(reflection.klass.cached_primary_index, ids_to_parent_record.keys) do |child_records_by_id|
+              parent_record_to_child_records = Hash.new { |h, k| h[k] = [] }
 
-          child_records = reflection.klass.fetch_multi(*ids_to_parent_record.keys)
-          child_records.each do |child_record|
-            parent_record = ids_to_parent_record[child_record.id]
-            parent_record_to_child_records[parent_record] << child_record
-          end
+              child_records_by_id.compact.each do |id, child_record|
+                next unless parent_record = ids_to_parent_record[id]
+                parent_record_to_child_records[parent_record] << child_record
+              end
 
-          parent_record_to_child_records.each do |parent, children|
-            parent.instance_variable_set(records_variable_name, children)
-          end
+              parent_record_to_child_records.each do |parent, children|
+                parent.instance_variable_set(records_variable_name, children)
+              end
 
-          child_records
+              yield child_records_by_id.values
```

Comment:
Doesn't this need to omit any `nil` values?  It looks like this was done since my last commit.

It looks like you repeated this in reference/has_one.rb as well.

Reply:
Right, I can probably get rid of some `nil` compactions / skipping if I omit them here. Also less likely to cache `nil` values this way I assume.

Reply:
> Also less likely to cache `nil` values this way I assume.

What gets cached is handled internally to the cache fetchers, so there shouldn't be a worry about caching the wrong thing outside of it.  Besides, it actually does cache nil for records that it doesn't find in the database.